src/v7400/connection/cache.rs

```diff
@@ -128,21 +129,20 @@ impl ConnectionsCacheBuilder {
                         && old_conn.label_sym() == conn.label_sym()
                 })
                 .expect("Should never fail: entry should exist");
-            return Err(ConnectionError::DuplicateConnection(
-                conn.source_id(),
-                conn.destination_id(),
-                conn.label_sym()
-                    .map(|sym| {
-                        self.labels.resolve(sym).expect(
-                            "Should never fail: connection label symbol should be registered",
-                        )
-                    })
-                    .map(ToOwned::to_owned),
-                old_conn.0,
-                old_conn.1.index(),
-                node.node_id(),
-                index,
-            ));
+
+            warn!(
+                    "Found duplicated connection node (node_id={:?}): \
+                    source_id={:?}, destination_id={:?}, label={:?}, first_conn={:?}",
+                    node.node_id(),
+                    conn.source_id(), conn.destination_id(), conn.label_sym()
+                        .map(|sym| {
+                             self.labels.resolve(sym).expect(
+                                 "Should never fail: connection label symbol should be registered",
+                             )
+                        })
+                        .map(ToOwned::to_owned),
+                    old_conn.0
+                );
         }
         self.connections.push((node.node_id(), conn));
```

Comment:
This line (and the following two lines) registers the duplicate connections as is, but is it meaningful?
While I'm not sure what the duplicate connections mean (I suspect that they are simply emitted by exporters bug), ignoring the redundant connection would be fine. At least object traversal won't be degraded by ignoring them.